Check/Extensions/Unrecommended.php

```diff
@@ -0,0 +1,144 @@
+          $ret_val .= '<tr><td>' . implode('</td><td>', $row) . '</td></tr>';
+        }
+        $ret_val .= '</tbody>';
+        $ret_val .= '</table>';
+      }
+      else {
+        foreach ($this->registry['extensions_unrec'] as $row) {
+          $ret_val .= PHP_EOL;
+          if (!drush_get_option('json')) {
+            $ret_val .= str_repeat(' ', 6);
+          }
+          $ret_val .= '- ' . $row[0] . ': ' . $row[1];
+        }
+      }
+    }
+    return $ret_val;
+  }
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getResultInfo().
+   */
+  public function getResultInfo() {}
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getResultPass().
+   */
+  public function getResultPass() {
+    return dt('No unrecommended extensions were detected; no action required.');
+  }
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getResultWarn().
+   */
+  public function getResultWarn() {}
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getAction().
+   */
+  public function getAction() {
+    if ($this->getScore() != SiteAuditCheckAbstract::AUDIT_CHECK_SCORE_PASS) {
+      return dt('Disable and completely remove unrecommended modules from your codebase for increased performance, stability and security in the any environment.');
+    }
+  }
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\calculateScore().
+   */
+  public function calculateScore() {
+    $this->registry['extensions_unrec'] = array();
+    $extension_info = $this->registry['extensions'];
+    uasort($extension_info, '_drush_pm_sort_extensions');
+    $unrecommended_extensions = $this->getExtensions();
+
+    foreach ($extension_info as $extension) {
+      $row = array();
+
+      $machineName = $extension->getName();
+
+      //get the human readable name of the module
+      if($extension->getType() == "module") {
+        $humanReadableName = \Drupal::moduleHandler()->getName($machineName);
+      }
+      //Cannot get the name of uninstalled themes. Needs to be fixed
+      /*else if($extension->getType() == "theme") {
+        $humanReadableName = \Drupal::service('theme_handler')->getName($machineName);
+      }*/
+      else {
+        $humanReadableName = $machineName;
+      }
+
+      //construct label from machine name and human readable name of the module
+      $label = $humanReadableName . '(' . $machineName . ')';
+      // Not in the list of known unrecommended modules.
+      if (!array_key_exists($machineName, $unrecommended_extensions)) {
+        continue;
+      }
+
+      // Name.
+      $row[] = $label;
+      // Reason.
+      $row[] = $unrecommended_extensions[$machineName];
+
+      $this->registry['extensions_unrec'][$machineName] = $row;
+    }
+
+    if (!empty($this->registry['extensions_unrec'])) {
+      return SiteAuditCheckAbstract::AUDIT_CHECK_SCORE_FAIL;
+    }
+    return SiteAuditCheckAbstract::AUDIT_CHECK_SCORE_PASS;
+  }
+
+  /**
+   * Get a list of unrecommended extension names and reasons.
```

Comment:
I have read the documentation for comments, but cannot figure out what's wrong with this line.